plugin/flags_sources/cmake_file.py

```diff
@@ -41,7 +41,8 @@ def __init__(self,
                  include_prefixes,
                  prefix_paths,
                  flags,
-                 cmake_binary="cmake"):
+                 cmake_binary="cmake",
+                 header_to_source_mapping=["{stamp}.*"]):
```

Comment:
nit: Do we need a default value over here? This value is not configurable and I feel it would be better to rely purely on the one defined in settings.

Reply:
I initially had an empty list here, but I think somewhere in the unit tests the CMake flags source is instantiated without passing in the default configuration values.

The default value here basically implements the previous lookup strategy for header files (i.e. if the file is not directly part of the DB, try to find a file with the same basename). This behavior is required by some of the unit tests.

To avoid the default value here, the compilation DB flag source could be extended to always check for a file with the same basename as last resort before using the merges flags.

Initially I planned to keep the previous implementation as hardwired fallback in, but as I had to change the key format in the DB (i.e. keys are now the full file names including the suffix) I did not want to introduce another few lines of code that would do the same as users would anyway get with the default settings now ;-)

Reply:
I completely support moving on from the old way. That was a hack that has survived surprisingly long :) 

I feel like we could move this before the cmake option, and make is always read the value from settings. I can adapt the tests when merging this if there are no other issues here.

Reply:
Okay, fine. Maybe I also got something wrong (I initially had some trouble running the Sublime unit tests but as that works now I think maybe I'll also find a better solution).

Reply:
Don't hesitate to leave it as is if it gets too annoying. It is not an issue for me to merge it the way I want to doing a couple of small changes on this.